subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -1692,20 +1692,25 @@ static void *radio_ccm_ext_rx_pkt_set(struct ccm *cnf, uint8_t phy, uint8_t pdu_
 		mode |= (CCM_MODE_DATARATE_1Mbit <<
 			 CCM_MODE_DATARATE_Pos) &
 			CCM_MODE_DATARATE_Msk;
+
+		if (false) {
+
 #if defined(CONFIG_BT_CTLR_DF_CONN_CTE_RX)
-		/* When direction finding CTE receive feature is enabled then on-the-fly PDU
-		 * parsing for CTEInfo is always done. In such situation, the CCM TASKS_CRYPT
-		 * must be started with short delay. That give the Radio time to store received bits
-		 * in shared memory.
-		 */
-		radio_bc_configure(CCM_TASKS_CRYPT_DELAY_BITS);
-		radio_bc_status_reset();
-		hal_trigger_crypt_by_bcmatch_ppi_config();
-		hal_radio_nrf_ppi_channels_enable(BIT(HAL_TRIGGER_CRYPT_DELAY_PPI));
-#else
-		hal_trigger_crypt_ppi_config();
-		hal_radio_nrf_ppi_channels_enable(BIT(HAL_TRIGGER_CRYPT_PPI));
+		} else if (pdu_type == RADIO_PKT_CONF_PDU_TYPE_DC) {
+			/* When direction finding CTE receive feature is enabled then on-the-fly
+			 * PDU parsing for CTEInfo is always done. In such situation, the CCM
+			 * TASKS_CRYPT must be started with short delay. That give the Radio time
+			 * to store received bits in shared memory.
+			 */
+			radio_bc_configure(CCM_TASKS_CRYPT_DELAY_BITS);
+			radio_bc_status_reset();
+			hal_trigger_crypt_by_bcmatch_ppi_config();
+			hal_radio_nrf_ppi_channels_enable(BIT(HAL_TRIGGER_CRYPT_DELAY_PPI));
```

Comment:
Hmm, are any of these unavailable when `CONFIG_BT_CTLR_DF_CONN_CTE_RX=n`? 

I am just wondering if the `#if defined(CONFIG_BT_CTLR_DF_CONN_CTE_RX)` could be replaced by a `if (IS_ENABLED)` which would make this look better

Reply:
`CCM_TASKS_CRYPT_DELAY_BITS` and `HAL_TRIGGER_CRYPT_DELAY_PPI` both are only defined under `CONFIG_BT_CTLR_DF_CONN_CTE_RX`

Reply:
Ah. What purpose is there to guard those defines?

Reply:
`HAL_TRIGGER_CRYPT_DELAY_PPI`  PPI instance/resource would then either be available or unavailable for application use:

https://github.com/zephyrproject-rtos/zephyr/blob/4d598683972a3601c2e90b53d58ffbb6e33e99ea/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/nrfx_glue/bt_ctlr_used_resources.h#L108-L113

Reply:
Is that a problem? 

At least in the host we rarely (if ever) guard `#define`s, as they do no harm.

Reply:
I have not checked if that is a problem (does not seem so), but the defines (header file using) are used externally (nrfx module) and will need more change/testing.